Services/Simulations.cs

```diff
@@ -35,7 +35,7 @@ public interface ISimulations
         /// <summary>
         /// Create or Replace a simulation.
         /// </summary>
-        Task<Models.Simulation> UpsertAsync(Models.Simulation simulation);
+        Task<Models.Simulation> UpsertAsync(Models.Simulation simulation, bool updatePartitioningCompleteFlag = false);
```

Comment:
this signature is really odd... upsert should be just upsert, without exceptions. The code used to work in the PoC, what changed?